Cesium.CodeGen/Contexts/AssemblyContext.cs

```diff
@@ -101,6 +107,41 @@ public AssemblyDefinition VerifyAndGetAssembly()
         {
             GlobalType = Module.GetType("<Module>");
         }
+        _globalTypeStaticCtor = new(
+            () =>
+            {
+                var ctor = BuildStaticCtor();
+                GlobalType.Methods.Add(ctor);
+                return ctor;
+            });
+    }
+
+    private MethodDefinition BuildStaticCtor()
+    {
+        var methodAttributes = MethodAttributes.Private
+                               | MethodAttributes.HideBySig
+                               | MethodAttributes.SpecialName
+                               | MethodAttributes.RTSpecialName
+                               | MethodAttributes.Static;
+        MethodDefinition method = new MethodDefinition(".cctor", methodAttributes, Module.TypeSystem.Void);
+        method.Body.Instructions.Add(Instruction.Create(OpCodes.Ret));
+        return method;
+    }
+
+    internal void AddFieldInitialization(FieldDefinition field, Ir.Expressions.IExpression value)
+    {
+        var lval = new Ir.Expressions.LValues.LValueGlobalVariable(field);
+        var context = new TranslationUnitContext(this);
+        var scope = new FunctionScope(context, GlobalTypeStaticCtor);
```

Comment:
Nope, you shouldn't create these objects here.

Instead of this, the caller should pass a `TranslationUnitContext`, and it should implement `IDeclarationScope` (so you'll be able to pass it to `EmitSetValue`).

Usage of `LValueGlobalVariable` is odd but okay.